### ISSUE TITLE
Gurobi and Cplex shared libraries not found

### ISSUE THREAD

**Issue by ThomasBasic** [casadi/casadi#2440]
Hi there, 
I am trying to solve a MIQP problem with either gurobi or cplex. Unfortunately, in both cases the same error occurs. For cplex, it looks like: 

> `CasADi - 2019-05-20 14:08:44 WARNING(".../casadi/core/plugin_interface.hpp:292: Assertion "handle!=nullptr" failed:
> PluginInterface::load_plugin: Cannot load shared library 'libcasadi_conic_cplex.so': 
>    (
>     Searched directories: 1. casadipath from GlobalOptions
>                           2. CASADIPATH env var
>                           3. PATH env var (Windows)
>                           4. LD_LIBRARY_PATH env var (Linux)
>                           5. DYLD_LIBRARY_PATH env var (osx)
>     A library may be 'not found' even if the file exists:
>           * library is not compatible (different compiler/bitness)
>           * the dependencies are not found
>    )
>   Tried '/local/programme/casadi-linux-matlabR2014b-v3.4.5' :
>     Error code: libcplex1263.so: Kann die Shared-Object-Datei nicht öffnen: Datei oder Verzeichnis nicht gefunden
>   Tried '' :
>     Error code: libcplex1263.so: Kann die Shared-Object-Datei nicht öffnen: Datei oder Verzeichnis nicht gefunden
>   Tried '.' :
>     Error code: ./libcasadi_conic_cplex.so: Kann die Shared-Object-Datei nicht öffnen: Datei oder Verzeichnis nicht gefunden") [.../casadi/core/plugin_interface.hpp:148]
> 
> Error using casadi.qpsol (line 8)
> .../casadi/core/plugin_interface.hpp:387: Plugin 'cplex' is not found.`

I am using: 
- Linux 64 bit
- Matlab 2018a 64 bit
- Casadi 3.4.5, installed as binary
- cplex: CPLEX_Studio_Community129 (x86-64_linux)
- gurobi: gurobi811 linux64

I did add both paths for cplex and gurobi to the PATH env variable in my .bashrc .

Does Casadi work with the newest versions of cplex and gurobi at all? Or do I need to get older versions of them? 

Thanks in advance, 
Thomas

**Comment by jgillis** [comment]
Current binaries needs cplex 1263 and gurobi 650.
A source build does not have those retrictions..

**Comment by ThomasBasic** [comment]
Thanks for the quick response! It took a while, but it worked. 

For future documentation: 
 - I needed to add a line `gurobi81` in `casadi/cmake/FindGUROBI.cmake` to make it work
```
find_library( GUROBI_LIBRARY 
              NAMES gurobi
              gurobi65
              gurobi70
              gurobi81
              PATHS "$ENV{GUROBI_HOME}/lib" 
                    "/Library/gurobi650/mac64/lib"
                    "C:\\libs\\gurobi650\\lib"
              )
```
- I needed to define environment variables GUROBI_HOME, CPLEX and MATLAB_ROOT (since my matlab install is not where it usually is). 
- For cmake, I used additional arguments: 
`cmake -DWITH_PYTHON=ON -DWITH_MATLAB=ON -DWITH_DEEPBIND=ON -DWITH_GUROBI=ON -DWITH_CPLEX=ON ..
`

**Comment by lalajiayou** [comment]
Hi, everybody.
I'm now encountering the same problem, and I want to call the Gurobi academic solver. There is a FAQ: how to get third party solvers to work? (at https://github.com/casadi/casadi/wiki/FAQ:-how-to-get-third-party-solvers-to-work%3F), and I've made some attempts, but I still can't make it happen. I wonder if it is convenient for you to provide some help.
Looking forward to your reply!

I am using:
Windows 64 bit
Matlab 2019b
Casadi 3.5.5, installed as binary
Cplex 12.9 (Academic Edition)
Gurobi 9.1.2 (Academic Edition)

By the way, I did add both paths for cplex and gurobi to the PATH env variable, that is, I can correctly call CPLEX and Gurobi to solve optimization problems.

Thanks in advance.
